### PR TITLE
Set plugins to be SHARED libraries

### PR DESCRIPTION
**Pull request by mpowelson**

### CODE REVIEW

**Comment by mpowelson** [comment]
@Levi-Armstrong This is apparently needed?

File: tesseract_ros/tesseract_plugins/CMakeLists.txt [hunk 1]
```diff
@@ -10,7 +10,7 @@ find_package(tesseract_collision REQUIRED)
 
 list(FIND CMAKE_CXX_COMPILE_FEATURES cxx_std_11 CXX_FEATURE_FOUND)
 
-add_library(tesseract_collision_bullet_plugin src/plugins/bullet_plugin.cpp)
+add_library(tesseract_collision_bullet_plugin SHARED src/plugins/bullet_plugin.cpp)
```
Comment (gavanderhoorn) [inline tesseract_ros/tesseract_plugins/CMakeLists.txt]: You may want to consider following CMake best-practices and using [BUILD_SHARED_LIBS](https://cmake.org/cmake/help/v3.5/variable/BUILD_SHARED_LIBS.html) instead.

Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_plugins/CMakeLists.txt]: Yea that would be good. I have seen the snippet below used to default to ON. Do you see any issues with this?

```
if(NOT DEFINED BUILD_SHARED_LIBS)
    set(BUILD_SHARED_LIBS ON)
endif()
```

Comment (gavanderhoorn) [reply tesseract_ros/tesseract_plugins/CMakeLists.txt]: No, that's how I do it as well. It's nice to give users the option, but still set a default.

Comment (Levi-Armstrong) [reply tesseract_ros/tesseract_plugins/CMakeLists.txt]: I created an issue #8

**Pull request merged**